src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/MiscStatement.scala

```diff
@@ -43,7 +43,7 @@ private[querygeneration] object MiscStatement {
     Option(expr match {
       case Alias(child: Expression, name: String) =>
         blockStatement(convertStatement(child, fields), name)
-      case Cast(child, t, _) =>
+      case Cast(child, t, _, _) =>
```

Comment:
For the interface change, we need to analysis:
1. What's the new parameter.
2. Can it be supported by snowflake corresponding function?

If it can be supported by snowflake, we will support it. Otherwise, we don't pushdown it.

Reply:
I mentioned it in my previous comment it is 'ansiEnabled' parameter, which was previously accessible through SQLConf. It is not new, only the way it is accessed has changed: https://github.com/apache/spark/commit/6f51e37eb52f21b50c8d7b15c68bf9969fee3567 ansi support is expermental feature on spark, as I see it throws exceptions, when overflow on some numeric operations happens, and restricts some keywords: https://spark.apache.org/docs/3.2.0/sql-ref-ansi-compliance.html as I understand, snowflake supports ansi dialect by default, should we think of a way to change behaviour, when ansi is disabled? Return nulls instead of overflow exceptions, etc...?

Reply:
If this is the case, you may need to investigate more:
1. Does snowflake support `ansiEnabled = true` and `ansiEnabled = false`, if so, `ansiEnabled` should be passed to snowflake.
2. If no, does the snowflake CAST to be aligned with ansiEnabled or not, we may only pushdown for one case. For example, 
`case Cast(child, t, _, ansiEnabled) if ansiEnabled =>` 
or 
`case Cast(child, t, _, ansiEnabled) if !ansiEnabled =>`

Reply:
Thanks for taking time to answer.

In this case, default behaviour on Snowflakes side should be pushing down `TRY_CAST` (https://docs.snowflake.com/en/sql-reference/functions/try_cast.html)
Which returns null, when cast fails, and using simple `CAST`, when ansiEnabled

That would align with statement from spark documentation:
> When spark.sql.ansi.enabled is set to true, Spark SQL uses an ANSI compliant dialect instead of being Hive compliant. For example, Spark will throw an exception at runtime instead of returning null results if the inputs to a SQL operator/function are invalid.

Would you agree on that?